.github/workflows/dependabot-automation.yml

```diff
@@ -0,0 +1,21 @@
+name: Dependabot automation
+
+on: pull_request
```

Comment:
This will run as soon as the PR is opened, yeah? Maybe let's wait until CI passes for it so we don't auto-merge broken stuff?

Reply:
> This will run as soon as the PR is opened, yeah?

Yeah good point, I overlooked this while testing.

> Maybe let's wait until CI passes for it so we don't auto-merge broken stuff?

I thought of 2 ways to do this:

1. Use GitHub's [auto-merge](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/automatically-merging-a-pull-request) feature. This requires enabling the following features:
    1. **Require status checks to pass before merging** under the repository's `master` branch protection rule
    2. **Allow auto-merge** under the repository's general settings
2. Trigger on `workflow_run` instead. This is what I went with in 13fbd76ed86d5a29bb9f2eb95bb195aae8549a10. It looks slightly hacky since `dependabot/fetch-metadata` [can no longer be used](https://github.com/dependabot/fetch-metadata/blob/aea2135c95039f05c64436f1d14638c300e10b2b/src/dependabot/verified_commits.ts#L16-L17), but avoids the implications of (1).

Reply:
A slight downside to (2): it is not obvious which workflow run is doing the auto-merging whereas previously the `pull_request` trigger allows the workflow run to show on the commit.

`pull_request` trigger:

![image](https://user-images.githubusercontent.com/13424970/227319496-e1ad95db-055b-415d-976d-df9c37d01fd0.png)

`workflow_run` trigger:

![image](https://user-images.githubusercontent.com/13424970/227319604-ebcdce0c-c0aa-4a7c-98a0-7438be00a8e9.png)

Reply:
Oh smart, why didn't I think of that... would've saved me a morning of tinkering with GitHub Actions. I'll try it out on the fork and update this PR.

Reply:
Done: 91355967558fc8101c2c8a98b7cb64893ed35da0

Tested over at https://github.com/victorlin/nextstrain.org/pull/4.